File: siaqodb/Documents/Indexes/IIndex.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sqo.Documents.Indexes
{
    interface IIndex : IDisposable
    {
        List<string> FindItem(object key);
        List<string> FindAllExcept(object key);
        List<string> FindItemsBetween(object start, object end);
        List<string> FindItemsBetweenExceptStart(object start, object end);
        List<string> FindItemsBetweenExceptEnd(object start, object end);
        List<string> FindItemsBetweenExceptStartEnd(object start, object end);
        List<string> FindItemsBiggerThan(object start);
        List<string> FindItemsBiggerThanOrEqual(object start);
        List<string> FindItemsLessThan(object start);
        List<string> FindItemsLessThanOrEqual(object start);
       
       
    }
}

File: Android/sqo/src/main/java/dotissi/sqo/IIndex.java
package dotissi.sqo;

import java.io.Closeable;
import java.util.ArrayList;

interface IIndex extends Closeable
        {
        ArrayList<String> FindItem(Object key);
        ArrayList<String> FindAllExcept(Object key);
        ArrayList<String> FindItemsBetween(Object start, Object end);
        ArrayList<String> FindItemsBetweenExceptStart(Object start, Object end);
        ArrayList<String> FindItemsBetweenExceptEnd(Object start, Object end);
        ArrayList<String> FindItemsBetweenExceptStartEnd(Object start, Object end);
        ArrayList<String> FindItemsBiggerThan(Object start);
        ArrayList<String> FindItemsBiggerThanOrEqual(Object start);
        ArrayList<String> FindItemsLessThan(Object start);
        ArrayList<String> FindItemsLessThanOrEqual(Object start);
        ArrayList<String> FindItemsStartsWith(Object target_key);
        ArrayList<String> FindItemsContains(Object target_key);
        ArrayList<String> FindItemsEndsWith(Object target_key);



        }